packages/sd-jwt/src/action-handler.ts

```diff
@@ -0,0 +1,227 @@
+    })
+    const credential = await sdjwt.present(
+      args.presentation,
+      args.presentationFrame as PresentationFrame<SdJwtVcPayload>,
+      { kb: args.kb },
+    )
+    return { presentation: credential }
+  }
+
+  /**
+   * Verify a signed SD-JWT credential.
+   * @param args - Arguments necessary for the verify a SD-JWT credential.
+   * @param context - This reserved param is automatically added and handled by the framework, *do not override*
+   * @returns
+   */
+  async verifySdJwtVc(args: IVerifySdJwtVcArgs, context: IRequiredContext): Promise<IVerifySdJwtVcResult> {
+    // biome-ignore lint/style/useConst: <explanation>
+    let sdjwt: SDJwtVcInstance
+    const verifier: Verifier = async (data: string, signature: string) =>
+      this.verify(sdjwt, context, data, signature)
+
+    sdjwt = new SDJwtVcInstance({ verifier, hasher: this.algorithms.hasher })
+    const verifiedPayloads = await sdjwt.verify(args.credential)
+
+    return { verifiedPayloads }
+  }
+
+  /**
+   * Verify the key binding of a SD-JWT by validating the signature of the key bound to the SD-JWT
+   * @param sdjwt - SD-JWT instance
+   * @param context - This reserved param is automatically added and handled by the framework, *do not override*
+   * @param data - signed data
+   * @param signature - The signature
+   * @param payload - The payload of the SD-JWT
+   * @returns
+   */
+  private verifyKb(sdjwt: SDJwtVcInstance, context: IRequiredContext, data: string, signature: string, payload: JwtPayload): Promise<boolean> {
+    if (!payload.cnf) {
+      throw Error('other method than cnf is not supported yet')
+    }
+    const key = payload.cnf.jwk as JsonWebKey
+    return this.algorithms.verifySignature(data, signature, key)
+  }
+
+  /**
+   * Validates the signature of a SD-JWT
+   * @param sdjwt - SD-JWT instance
+   * @param context - This reserved param is automatically added and handled by the framework, *do not override*
+   * @param data - signed data
+   * @param signature - The signature
+   * @returns
+   */
+  async verify(sdjwt: SDJwtVcInstance, context: IRequiredContext, data: string, signature: string) {
+    const decodedVC = await sdjwt.decode(`${data}.${signature}`)
+    const issuer: string = ((decodedVC.jwt as Jwt).payload as Record<string, unknown>).iss as string
+    if (!issuer.startsWith('did:')) {
+      throw new Error('invalid_issuer: issuer must be a did')
+    }
+    const didDoc = await context.agent.resolveDid({ didUrl: issuer })
+    if (!didDoc) {
+      throw new Error('invalid_issuer: issuer did not resolve to a did document')
+    }
+    const didDocumentKey = didDoc.didDocument?.verificationMethod?.find((key) => key.id)
+    if (!didDocumentKey) {
+      throw new Error('invalid_issuer: issuer did document does not include referenced key')
+    }
+    //TODO: in case it's another did method, the value of the key can be also encoded as a base64url
+    const key = didDocumentKey.publicKeyJwk as JsonWebKey
```

Comment:
This needs more checking. Because the above TODO really should be a fixme. This for sure will give issues for some DID methods as they do not expose the keys as publicKeyJwk

Reply:
changed to a FIXME and added a comment. is this something that should be fixed now and what approach can we take to fix this?

Reply:
No we can skip it for now